src/sinol_make/helpers/compile.py

```diff
@@ -27,6 +64,14 @@ def compile(program, output, compilers: Compilers = None, compile_log = None, we
     if extra_compilation_files is None:
         extra_compilation_files = []
 
+    compiled_exe = check_compiled(program)
+    if compiled_exe is not None:
+        if compile_log is not None:
```

Comment:
Yes, I'm still forgetting that this is the better way for writing stuff like this

Reply:
What do you think about implementing one of those (or another one that is as good as those) in the whole codebase?

Reply:
Is there no obvious ways to use this in at least some of those following places?
```
src/sinol_make/util.py
37:def check_oiejq(path = None):
55:    if path is not None:
109:        return None
122:        return None
144:            "default_flow_style": None
148:            "default_flow_style": None
152:            "default_flow_style": None
189:def check_for_updates(current_version) -> Union[str, None]:
193:    :return: returns new version if there is one, None otherwise
212:                return None
214:            return None
216:        return None
338:def exit_with_error(text, func=None):
340:    if func is not None:

src/sinol_make/helpers/compile.py
27:    :return: executable path if compiled, None otherwise
39:                return None
41:        return None
57:def compile(program, output, compilers: Compilers = None, compile_log = None, weak_compilation_flags = False,
58:            extra_compilation_args = None, extra_compilation_files = None):
69:    if extra_compilation_args is None:
71:    if extra_compilation_files is None:
75:    if compiled_exe is not None:
76:        if compile_log is not None:
90:    if compilers is None:
123:    if compile_log is not None:
137:        -> Tuple[Union[str, None], str]:
144:    :return: Tuple of (executable path or None if compilation failed, log path)
160:        return None, compile_log_path

src/sinol_make/helpers/printer.py
15:                 Title and footer can be None.
19:    wrapper(_printer, None, func, *args, **kwargs)
30:                 Title and footer can be None.
44:                 Title and footer can be None.
63:    last_title = None
64:    last_footer = None
66:    title = None
67:    footer = None
70:        while run_event is None or run_event.is_set():
79:            if title is not None:
81:            if footer is not None:
107:                if title is not None:
109:                _print_to_scr(stdscr, '\n'.join(output[curr_row:curr_row + visible_height]), title is not None)
110:                if footer is not None:

src/sinol_make/helpers/package_util.py
29:def get_tests(arg_tests: Union[List[str], None]) -> List[str]:
32:    :param arg_tests: Tests specified in command line arguments. If None, all tests are returned.
35:    if arg_tests is None:

src/sinol_make/__init__.py
41:            if new_version is not None:

src/sinol_make/helpers/compiler.py
32:            return None
41:        return None
51:            return None
60:        return None
69:        return None
80:        return None
109:        if ext == '.c' and args.c_compiler_path is None:
116:        elif ext == '.cpp' and args.cpp_compiler_path is None:
123:        elif ext == '.py' and args.python_interpreter_path is None:
127:        elif ext == '.java' and args.java_compiler_path is None:

src/sinol_make/structs/compiler_structs.py
13:    def __init__(self, c_compiler_path: str = None, cpp_compiler_path: str = None, python_interpreter_path: str = None, java_compiler_path: str = None):

tests/fixtures.py
11:    if hasattr(request, 'param') and request.param is not None:

tests/helpers/test_compile.py
15:        assert compile.check_compiled(os.path.join(os.getcwd(), "test.txt")) is None

src/sinol_make/commands/inwer/inwer_util.py
16:def get_inwer_path(task_id: str, path = None) -> Union[str, None]:
18:    Returns path to inwer executable for given task or None if no inwer was found.
20:    if path is None:
23:            return None
29:        return None
106:    return new_stdout.getvalue().splitlines(), None, "Use arrows to move."

src/sinol_make/commands/inwer/__init__.py
45:        if self.inwer_executable is None:
128:        if self.inwer is None:
129:            if args.inwer_path is None:

src/sinol_make/commands/gen/__init__.py
70:        old_md5_sums = None
85:            if old_md5_sums is None or old_md5_sums.get(basename, '') != md5_sums[basename]:

src/sinol_make/commands/run/__init__.py
115:                    if getattr(results[test], "Time") is not None:
121:                    if getattr(results[test], "Memory") is not None:
176:        last_group = None
180:                if last_group is not None:
192:                         if getattr(result, "Time") is not None else 7*" ", end=" | ")
199:                          if getattr(result, "Memory") is not None else 10*" ", end=" | ")
296:        if args_solutions is None:
405:        if not hasattr(self, "checker") or self.checker is None:
460:        elif getattr(result, "Time") is not None and result.Time > time_limit:
462:        elif getattr(result, "Memory") is not None and result.Memory > memory_limit:
464:        elif getattr(result, "Status") is None:
508:        program_exit_code = None
532:        if program_exit_code is not None and program_exit_code != 0:
672:        print(yaml.dump(yaml_dict, default_flow_style=None))
699:        if self.checker is None:
714:        if self.args.solutions == None and config_expected_scores: # If no solutions were specified, use all solutions from config
723:        if self.args.tests == None and config_expected_scores: # If no groups were specified, use all groups from config
769:                if field == '' and self.args.solutions == None and config_expected_scores:
773:                elif field[1] == "expected" and self.args.tests == None and config_expected_scores:
896:        compilers = compiler.verify_compilers(args, self.get_solutions(None))
898:        timetool_path = None
902:            if 'oiejq_path' in args and args.oiejq_path is not None:
908:            if timetool_path is None:
1016:                    if results[solution][group][test].Status == "CE" and results[solution][group][test].Error is not None:
1047:        self.time_limit = args.tl * 1000.0 if args.tl is not None else config_time_limit
1048:        self.memory_limit = args.ml * 1024 if args.ml is not None else config_memory_limit
1072:            self.checker = None

src/sinol_make/commands/gen/gen_util.py
13:def get_ingen(task_id=None, ingen_path=None):
17:    :param task_id: task id, for example abc. If None, then
20:    :return: path to ingen source file or None if not found
23:    if ingen_path is not None:
29:    if task_id is None:
49:    if ingen_exe is None:
60:    :return: path to correct solution or None if not found
75:    if correct_solution_exe is None:
99:    while process.poll() is None:
```

Reply:
I made a few changes

Reply:
Let's continue this conversation on Zulip.